### PR TITLE
enable nil time event processing in usps

### PR DESCRIPTION
**Pull request by adamjk**

### CODE REVIEW

File: lib/active_shipping/shipping/carriers/usps.rb [hunk 1]
```diff
@@ -578,22 +582,25 @@ def parse_tracking_with_fields_response(response)
...
+          status, status_category, expected_delivery_date, actual_delivery_date = nil, nil, nil, nil
 
           if success 
+            previous_event_time = Time.now
             tracking_details.each do |event|
-              ship_event = extract_track_with_fields_event(event)
-              shipment_events << ship_event if !ship_event.nil?
+              ship_event = extract_track_with_fields_event(event, previous_event_time)
+              previous_event_time = ship_event.time
+              shipment_events << ship_event
```
Comment (sslotnick) [inline lib/active_shipping/shipping/carriers/usps.rb:592]: Looks like you dropped the if nil check, is it now impossible for ship_event to be nil?

File: lib/active_shipping/shipping/carriers/usps.rb [hunk 2]
```diff
@@ -578,22 +582,25 @@ def parse_tracking_with_fields_response(response)
...
             if !summary_ship_event.nil?
               shipment_events << summary_ship_event
+              status = summary_ship_event.status
               actual_delivery_date = summary_ship_event.time if summary_ship_event.delivered?
             end
 
             shipment_events = shipment_events.sort_by(&:time)
-            status = tracking_info.elements['StatusCategory'].get_text.to_s
-            status = status.downcase.gsub("\s", "_").to_sym
+            status_category = tracking_info.elements['StatusCategory'].get_text.to_s
```
Comment (sslotnick) [inline lib/active_shipping/shipping/carriers/usps.rb:602]: Was this just a previous bug in the library? What is the difference between the status and status category?

**Comment by sslotnick** [comment]
2 questions, otherwise looks good

File: lib/active_shipping/shipping/carriers/usps.rb (hunk 1 above)
Comment (adamjk) [inline lib/active_shipping/shipping/carriers/usps.rb:592]: yeah, its not possible for it to be nil

File: lib/active_shipping/shipping/carriers/usps.rb (hunk 2 above)
Comment (adamjk) [inline lib/active_shipping/shipping/carriers/usps.rb:602]: It's not a bug. I previosuly used the 'status category' from USPS, which is specific to that carrier, as the status. It doesn't very well represent the status as I've found out, it's more coarse. So this adds the status category and uses the current status element in the USPS response for the status.

**Pull request merged**